tests/test_tacacs_server_group.rb

```diff
@@ -17,98 +14,392 @@
-    assert_equal(['8.8.8.8', '10.10.10.10'],
-                 Cisco::TacacsServerGroup.tacacs_server_groups[id].servers)
-    group2.servers = ['11.11.11.11', '13.13.13.13']
-    assert_equal(['11.11.11.11', '13.13.13.13'],
-                 Cisco::TacacsServerGroup.tacacs_server_groups[id2].servers)
-
-    group.servers = []
-    assert_equal([],
-                 Cisco::TacacsServerGroup.tacacs_server_groups[id].servers)
-    group2.servers = []
-    assert_equal([],
-                 Cisco::TacacsServerGroup.tacacs_server_groups[id2].servers)
-
-    group.destroy
-    group2.destroy
-    refute_includes(Cisco::TacacsServerGroup.tacacs_server_groups, id)
-    refute_includes(Cisco::TacacsServerGroup.tacacs_server_groups, id2)
+  def clean_tacacs_config
+    config('no feature tacacs',
+           'feature tacacs')
+  end
+
+  def create_tacacsserverhost(name='defaulttest')
+    TacacsServerHost.new(name)
+  end
+
+  def detach_tacacsserverhost(host)
+    host.destroy
+  end
+
+  def detach_aaaservergroup(aaa_server_group)
+    aaa_server_group.destroy
+  end
+
+  def create_aaa_group(group_name, server)
+    config("aaa group server #{server} #{group_name}")
+  end
+
+  def destroy_aaa_group(group_name, server)
+    config("no aaa group server #{server} #{group_name}")
+  end
+
+  def create_vrf(group_name, server, vrf_name)
+    config("aaa group server #{server} #{group_name} ; use-vrf #{vrf_name}")
+  end
+
+  def create_deadtime(group_name, server, deadtime)
+    config("aaa group server #{server} #{group_name} ; deadtime #{deadtime}")
+  end
+
+  def create_source_interface(group_name, server, interface)
+    config("aaa group server #{server} #{group_name} ; " \
+           "source-interface #{interface}")
+  end
+
+  def test_create_invalid_name_tacacs
+    assert_raises(TypeError) do
+      TacacsServerGroup.new(nil)
+    end
+  end
+
+  def test_create_valid_tacacs
+    group_name = 'Group1'
+    aaa_group = TacacsServerGroup.new(group_name)
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /#{group_name}/)
+
+    detach_aaaservergroup(aaa_group)
+  end
+
+  def test_create_valid_multiple_tacacs
+    group_name1 = 'Group1'
+    group_name2 = 'Group2'
+    aaa_group1 = TacacsServerGroup.new(group_name1)
+    aaa_group2 = TacacsServerGroup.new(group_name2)
+
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /#{group_name1}/)
+    assert_show_match(command: 'show run tacacs+ all | no-more',
+                      pattern: /#{group_name2}/)
+
+    detach_aaaservergroup(aaa_group1)
+    detach_aaaservergroup(aaa_group2)
+  end
+
+  def test_collection_empty_tacacs
+    clean_tacacs_config
+    aaa_group_list = TacacsServerGroup.groups
+    assert_empty(aaa_group_list,
+                 'Error: TacacsServerGroup collection is not empty')
+  end
+
+  def test_collection_multi_tacacs
+    clean_tacacs_config
+    group_name1 = 'Group1'
```

Comment:
Why not make these global so you don't have to declare them in each test method.